src/sdk/pynni/nni/compression/torch/builtin_pruners.py

```diff
@@ -106,3 +106,124 @@ def update_epoch(self, epoch):
             self.now_epoch = epoch
             for k in self.if_init_list:
                 self.if_init_list[k] = True
+
+class FPGMPruner(Pruner):
+    """
+    A filter pruner via geometric median.
+    "Filter Pruning via Geometric Median for Deep Convolutional Neural Networks Acceleration",
+    https://arxiv.org/pdf/1811.00250.pdf
+    """
+
+    def __init__(self, model, config_list):
+        """
+        Parameters
+        ----------
+        model : pytorch model
+            the model user wants to compress
+        config_list: list
+            support key for each list item:
+                - pruning_rate: percentage of convolutional filters to be pruned.
+        """
+        super().__init__(model, config_list)
+        self.mask_list = {}
+
+    def calc_mask(self, layer, config):
+        """
+        Supports Conv1d, Conv2d
+        filter dimensions for Conv1d:
+        OUT: number of output channel
+        IN: number of input channel
+        LEN: filter length
+
+        filter dimensions for Conv2d:
+        OUT: number of output channel
+        IN: number of input channel
+        H: filter height
+        W: filter width
+
+        Parameters
+        ----------
+        layer : LayerInfo
+            calculate mask for `layer`'s weight
+        config : dict
+            the configuration for generating the mask
+        """
+        weight = layer.module.weight.data
+        assert 0 <= config.get('pruning_rate') < 1
+        assert layer.type in ['Conv1d', 'Conv2d']
+        assert layer.type in config['op_types']
+
+        if layer.name in self.epoch_pruned_layers:
+            assert layer.name in self.mask_list
+            return self.mask_list.get(layer.name)
+
+        masks = torch.ones(weight.size())
```

Comment:
use `type_as` to make mask placed at the same device with the corresponding weight

Reply:
Thanks, updated